src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionSource_ISuggestedActionsSourceExperimental.cs

```diff
@@ -0,0 +1,121 @@
+                ISuggestedActionCategorySet requestedActionCategories,
+                Document document,
+                SnapshotSpan range,
+                Func<string, IDisposable?> addOperationScope,
+                CodeActionRequestPriority priority,
+                [EnumeratorCancellation] CancellationToken cancellationToken)
+            {
+                var workspace = document.Project.Solution.Workspace;
+                var supportsFeatureService = workspace.Services.GetRequiredService<ITextBufferSupportsFeatureService>();
+
+                var selection = TryGetCodeRefactoringSelection(state, range);
+
+                var fixesTask = GetCodeFixesAsync(
+                    state, supportsFeatureService, requestedActionCategories, workspace, document, range,
+                    addOperationScope, priority, isBlocking: false, cancellationToken);
+                var refactoringsTask = GetRefactoringsAsync(
+                    state, supportsFeatureService, requestedActionCategories, workspace, document, selection,
+                    addOperationScope, priority, isBlocking: false, cancellationToken);
+
+                if (priority == CodeActionRequestPriority.High)
+                {
+                    // in a high pri scenario, return data as soon as possible so that the user can interact with them.
+                    // this is especially important for state-machine oriented refactorings (like rename) where the user
+                    // should always have access to them effectively synchronously.
+                    var firstTask = await Task.WhenAny(fixesTask, refactoringsTask).ConfigureAwait(false);
+                    var secondTask = firstTask == fixesTask ? refactoringsTask : fixesTask;
+
+                    var orderedTasks = new[] { firstTask, secondTask };
+                    foreach (var task in orderedTasks)
+                    {
+                        if (task == fixesTask)
+                        {
+                            var fixes = await fixesTask.ConfigureAwait(false);
+                            foreach (var set in ConvertToSuggestedActionSets(state, selection, fixes, ImmutableArray<UnifiedSuggestedActionSet>.Empty))
+                                yield return set;
+                        }
+                        else
+                        {
+                            Contract.ThrowIfFalse(task == refactoringsTask);
+
+                            var refactorings = await refactoringsTask.ConfigureAwait(false);
+                            foreach (var set in ConvertToSuggestedActionSets(state, selection, ImmutableArray<UnifiedSuggestedActionSet>.Empty, refactorings))
+                                yield return set;
+                        }
+                    }
+                }
+                else
+                {
+                    var actionsArray = await Task.WhenAll(fixesTask, refactoringsTask).ConfigureAwait(false);
```

Comment:
yes.  very deliberate to still wait on this group.  Outside of hte high-pri group, we want the rest of hte actions to still be sorted in the same manner as today.  Because of that, we still need to wait until they're done and merge/sort the results before presenting htem.